scripts/ubuntu.sh

```diff
@@ -0,0 +1,223 @@
+
+# need to put libhdf5.so on LD_LIBRARY_PATH (Making sure that LD_LIBRARY_PATH is defined first)
+if [ -z $LD_LIBRARY_PATH ]; then
+  export LD_LIBRARY_PATH="${hdf5_libdir}"
+else
+  export LD_LIBRARY_PATH="${hdf5_libdir}:$LD_LIBRARY_PATH"
+fi
+
+
+############
+### MOAB ###
+############
+# pre-setup
+cd ${install_dir}
+check_repo moab
+mkdir -p moab
+cd moab
+
+# clone and version
+git clone --depth 1 --single-branch -b 5.3.0 https://bitbucket.org/fathomteam/moab moab-repo
+cd moab-repo
+mkdir -p build
+cd build
+
+# cmake, build and install
+cmake ../ -DENABLE_PYMOAB=ON \
+          -DCMAKE_INSTALL_PREFIX=${install_dir}/moab \
+          -DENABLE_HDF5=ON -DHDF5_ROOT=${hdf5_libdir} \
+          -DBUILD_SHARED_LIBS=ON \
+          -DENABLE_BLASLAPACK=OFF \
+          -DENABLE_FORTRAN=OFF    
+make -j 3 
+cd pymoab
+pip install .
+cd ..
+make install
+cd ..
+rm -rf moab-repo
+
+# Adding MOAB/lib to $LD_LIBRARY_PATH and $LIBRARY_PATH
+export LD_LIBRARY_PATH="${install_dir}/moab/lib:$LD_LIBRARY_PATH"
+
+# Adding pymoab to $PYTHONPATH
+PYTHON_VERSION=$(python -c 'import sys; print(sys.version.split('')[0][0:3])')
+if [ -z $PYTHONPATH ]; then
+  export PYTHONPATH="$install_dir/moab/lib/python${PYTHON_VERSION}/site-packages"
+else
+  export PYTHONPATH="$install_dir/moab/lib/python${PYTHON_VERSION}/site-packages:$PYTHONPATH"
+fi
+
+
+#############
+### DAGMC ###
+#############
+# pre-setup check that the directory we need are in place
+cd ${install_dir}
+check_repo dagmc
+mkdir -p dagmc
+cd dagmc
+
+# clone and version
+git clone --depth 1 --branch stable https://github.com/svalinn/DAGMC.git dagmc-repo
+cd dagmc-repo
+git checkout develop
+mkdir build
+cd build
+
+# cmake, build and install
+cmake .. -DMOAB_DIR=${install_dir}/moab \
+         -DCMAKE_INSTALL_PREFIX=${install_dir}/dagmc \
+         -DBUILD_STATIC_LIBS=OFF \
+         -DBUILD_UWUW=OFF \
+         -DBUILD_TALLY=OFF \
+         -DBUILD_MAKE_WATERTIGHT=OFF \
+         -DBUILD_OVERLAP_CHECK=OFF \
+         -DBUILD_TESTS=OFF 
+make -j 3
+make install
+cd ../..
+rm -rf dagmc-repo
+
+# Adding DAGMC/lib to $LD_LIBRARY_PATH and $LIBRARY_PATH
+export LD_LIBRARY_PATH="${install_dir}/dagmc/lib:$LD_LIBRARY_PATH"
+
+# Adding dagmc/bin to $PATH
+export PATH="${install_dir}/dagmc/bin:$PATH"
+
+####################
+### OpenMC API #####
+####################
+
+cd ${install_dir}
+git clone https://github.com/openmc-dev/openmc.git
+cd openmc
+git checkout tags/v0.13.0
+mkdir bld
+cd bld
+cmake .. -DCMAKE_INSTALL_PREFIX=$HOME/.local
+make
+make install
+cd ..
+pip3 install .
+
+############
+### PyNE ###
+############
+
+# pre-setup
+export PYNE_HDF5_ARGS="--hdf5 ${hdf5_libdir}"
+cd ${install_dir}
+check_repo pyne
+mkdir -p pyne
+cd pyne
+# clone and version
+git clone -b develop --single-branch https://github.com/pyne/pyne.git pyne-repo
+cd pyne-repo
+if [ $1 == 'stable' ] ; then
+  TAG=$(git describe --abbrev=0 --tags)
+  git checkout tags/`echo ${TAG}` -b `echo ${TAG}`
```

Comment:
Can we do this if we've only cloned a single branch?

Reply:
I don't think lines 176-178 are being run. In [this workflow run](https://github.com/bquan0/pyne/actions/runs/4960351409/jobs/8875715298), line 1034 corresponds to line 173 in `ubuntu.sh`, and the lines that follow in the workflow run don't seem to have any tags echoed.

Reply:
I think they are being run but the condition is never true.  If it were ever true, it would probably fail.  Since this is a sample script for users to use to install, we should choose a good practice.

I think this was to allow people to run this script to install a guaranteed stable version rather than the develop branch.  That's probably a good thing to preserve, and therefore we might want to test both versions

Reply:
After implementing the matrix idea for testing both the develop branch and the stable branch, it seems like it is possible to switch to the stable branch even though only a single branch was cloned. In line 924 of [this workflow run](https://github.com/bquan0/pyne/actions/runs/5150609113/jobs/9274941733) with a command line arg of `stable`, it says `Switched to a new branch '0.7.7'`. This line is not there in [the workflow run](https://github.com/bquan0/pyne/actions/runs/5150609113/jobs/9274941603) with a different command line arg (see line 929)